test/Microsoft.Extensions.CommandLineUtils.Tests/CommandLineApplicationTests.cs

```diff
@@ -501,5 +501,45 @@ public void NestedInheritedOptions()
             Assert.Equal("N1", nest1OptionValue);
             Assert.Equal("N2", nest2OptionValue);
         }
+
+        [Fact]
+        public void HelpTextIgnoresHiddenItems()
+        {
+            var app = new CommandLineApplication()
+            {
+                Name = "ninja-app",
+                Description = "You can't see it until it is too late"
+            };
+
+            app.Command("star", c =>
+            {
+                c.Option("--points <p>", "How many", CommandOptionType.MultipleValue);
+                c.Hidden = true;
+            });
+            app.Option("--no-kill", "Be a nice ninja", CommandOptionType.NoValue, o => { o.Hidden = true; });
```

Comment:
The terms pass my search on the db, but I can change it if there's any chance it'll cause trouble.